### PR TITLE
Make Her Rails 5.2 compatible

### PR DESCRIPTION
**Pull request by zoltankiss**
I've ran unit tests against 5.2, 5.1, 5.0, 4.2, 3.2.

### CODE REVIEW

File: lib/her/model/attributes.rb [hunk 1]
```diff
@@ -31,15 +31,25 @@ def initialize(attributes={})
 
         attributes = self.class.default_scope.apply_to(attributes)
         assign_attributes(attributes)
+        _clear_changes_information unless new?
         yield self if block_given?
         run_callbacks :initialize
       end
 
+      def _clear_changes_information
```
Comment (zacharywelch) [inline lib/her/model/attributes.rb]: We've actually been meaning to update activemodel for some time. 4.2.1 or 4.2.7 will give us the same [api](https://apidock.com/rails/ActiveModel/Dirty) and help clean up some of the existing code.

File: lib/her/model/attributes.rb [hunk 2]
```diff
@@ -69,48 +79,55 @@ def respond_to_missing?(method, include_private = false)
...
       def attributes
-        @attributes ||= HashWithIndifferentAccess.new
+        # The natural choice of instance variable naming here would be
+        # `@attributes`. Unfortunately that causes a naming clash when
+        # used with `ActiveModel` version >= 5.2.0.
+        # As of v5.2.0 `ActiveModel` checks to see if `ActiveRecord`
+        # attributes exist, assuming that if the instance variable
+        # `@attributes` exists on the instance, it is because they are
+        # `ActiveRecord` attributes.
+        @_attributes ||= HashWithIndifferentAccess.new
```
Comment (zacharywelch) [inline lib/her/model/attributes.rb]: Makes sense. We have other examples where instance variables follow a `@_<variable>` format but they're prefixed with `@_her_<variable>`. Not a strong justification but following that format would give us consistency.

**Comment by zacharywelch** [comment]
Just remembered we have another open PR #298 that cleans up change tracking. Let me see if it can be merged because it should help w/ this PR 👍

**Comment by zacharywelch** [comment]
@zoltankiss I started working on a [branch](https://github.com/remiprev/her/tree/zw/activemodel-update) which finally updates activemodel. Some of the changes introduced in 4.2.1 we depend on are backwards incompatible so our best option is probably to branch out the codebase like [Devise](https://github.com/plataformatec/devise/branches/all) and [Rails](https://github.com/rails/rails/branches/all). Once that's done we should have a better idea what's  required to support Rails 5.2. It looks like we already know `@attributes` is one of those things 😺

**Comment by zacharywelch** [comment]
@zoltankiss can we go ahead and create a separate PR for https://github.com/remiprev/her/pull/492/commits/b6bc6b16f21fea7f7cd9197f60cde26781a8e333 since it's backwards compatible? Once we branch the code we won't have to worry about keeping those changes in sync 😺

**Comment by zoltankiss** [comment]
@zacharywelch https://github.com/remiprev/her/pull/497

**Comment by edtjones** [comment]
Excellent news @grk 

@zoltankiss @zacharywelch are there known issues still to resolve?

**Comment by zacharywelch** [comment]
Only the changes to activemodel 4.2.1 described [above](https://github.com/remiprev/her/pull/492#issuecomment-381149397). We've accumulated some technical debt related to dirty attributes that we're trying to pay off in the process of supporting 5.2. Thanks for the extra help testing @grk! We should have a more robust solution ready for everyone by this Saturday 😺

**Comment by zacharywelch** [comment]
Gonna close this now that we've merged #500. For those interested, 5.2 support is provided in v1.0.0 🎉 

Thanks @zoltankiss!

**Comment by eduardofiorezi** [comment]
👏

**Pull request closed without merging**